pkg/ebpf/dns_cache.go

```diff
@@ -68,12 +75,14 @@ func (c *reverseDNSCache) Add(translation *translation, now time.Time) bool {
 		val, ok := c.data[addr]
 		if ok {
 			val.expiration = exp
-			val.merge(translation.dns)
-			continue
+			if rejected := val.merge(translation.dns, c.maxDomainsPerIP); rejected {
+				log.Warnf("%s mapped to too many domains, DNS information will be dropped (this will be logged the first 10 times, and then at most every 10 minutes)", addr)
+				break
```

Comment:
Should we break here? Is it possible that the next `addr` may still have space left for `translation.dns`?

Reply:
That's true. Done